erpnext/stock/doctype/pick_list/pick_list.py

```diff
@@ -41,6 +51,24 @@ def before_submit(self):
 			frappe.throw(_('For item {0} at row {1}, count of serial numbers does not match with the picked quantity')
 				.format(frappe.bold(item.item_code), frappe.bold(item.idx)), title=_("Quantity Mismatch"))
 
+	def before_cancel(self):
+		#update picked_qty in SO Item on cancel of PL
+		self.update_so()
+
+	def update_so(self):
+		for item in self.locations:
+			if item.sales_order_item:
+				frappe.db.set_value("Sales Order Item",item.sales_order_item,"picked_qty",item.picked_qty)
```

Comment:
What if same item picked multiple times in the different pick lists? For example in the sales order item's qty is 5 and use has made 2 pick list with 4 qty in the first pick list and 1 qty in the second pick list

Reply:
Added validation on submit of PL.